app/ui-react/packages/api/src/WithUserHelpers.tsx

```diff
@@ -0,0 +1,51 @@
+import * as React from 'react';
+import { ApiContext, IApiContext } from './ApiContext';
+import { callFetch } from './callFetch';
+
+export interface IWithUserHelpersChildrenProps {
+  logout(): Promise<any>;
+}
+export interface IWithUserHelpersProps {
+  children(props: IWithUserHelpersChildrenProps): any;
+}
+
+export class WithUserHelpersWrapped extends React.Component<
+  IWithUserHelpersProps & IApiContext
+> {
+  constructor(props: IWithUserHelpersProps & IApiContext) {
+    super(props);
+    this.logout = this.logout.bind(this);
+  }
+
+  public async logout() {
+    const response = await callFetch({
+      headers: {
+        ...this.props.headers,
+        responseType: 'arraybuffer',
+      },
+      method: 'GET',
+      url: `${this.props.apiUri}/logout`,
+    });
+    // console.log(response);
+    if (!response.ok) {
+      throw new Error(response.statusText);
+    }
+    return await response.body;
```

Comment:
So I think instead of doing `return await response.body` we're gonna want to do something like:

```
        const html = String.fromCharCode.apply(null, new Uint8Array(response.body));
        window.history.pushState(null, null, '/logout');
        window.document.open();
        window.document.write(html);
        window.document.close();
        return;
```

I think that should do the trick?

Reply:
Yea I've been trying to make this work but getting this type error. Any ideas?

<img width="1141" alt="Screen Shot 2019-05-22 at 7 39 04 AM" src="https://user-images.githubusercontent.com/5942899/58171856-f033a100-7c64-11e9-80d4-9097f13c4457.png">

Reply:
And the second one there for pushState gives the error `Argument of type 'null' is not assignable to parameter of type 'string'.` I just pushed the latest with what I have, will continue looking into these errors though.